File: adhoc/119_pascals_triangle_ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if rowIndex == 0: return [1]

        res = [1, 1]
        for i in range(2, rowIndex + 1):
            temp = []
            for j in range(len(res) - 1):
                temp.append(res[j] + res[j + 1])
            res = [1] + temp + [1]
        return res

File: recursion/119_pascals_triangle_ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        if (rowIndex == 0) return Arrays.asList(1);
        if (rowIndex == 1) return Arrays.asList(1, 1);
        
        List<Integer> above = getRow(rowIndex - 1);
        List<Integer> res = new ArrayList();
        res.add(1);
        
        for (int i = 0; i < above.size() - 1; i++) {
            res.add(above.get(i) + above.get(i+1));
        }
        res.add(1);
        return res;
    }
}